File: .claude/skills/fixtures/core6-mechanical/go/extract/before.go
package main

import (
	"errors"
	"fmt"
)

// CreateAccount creates a new user account after validating input.
func CreateAccount(username, email string, age int) (string, error) {
	// Validation block — candidate for extraction
	if username == "" {
		return "", errors.New("username cannot be empty")
	}
	if len(username) < 3 {
		return "", errors.New("username must be at least 3 characters")
	}
	if email == "" {
		return "", errors.New("email cannot be empty")
	}
	if age < 18 {
		return "", errors.New("must be at least 18 years old")
	}

	accountID := fmt.Sprintf("acct_%s", username)
	return accountID, nil
}

func main() {
	id, err := CreateAccount("alice", "alice@example.com", 25)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	fmt.Println("Created account:", id)
}

File: .claude/skills/fixtures/core6-mechanical/rust/extract/before.rs
fn create_account(username: &str, email: &str, age: u32) -> Result<String, String> {
    // Validation block — candidate for extraction
    if username.is_empty() {
        return Err("username cannot be empty".to_string());
    }
    if username.len() < 3 {
        return Err("username must be at least 3 characters".to_string());
    }
    if email.is_empty() {
        return Err("email cannot be empty".to_string());
    }
    if age < 18 {
        return Err("must be at least 18 years old".to_string());
    }

    Ok(format!("acct_{}", username))
}

fn main() {
    match create_account("alice", "alice@example.com", 25) {
        Ok(id) => println!("Created: {}", id),
        Err(e) => println!("Error: {}", e),
    }
}
